src/components/Sing/SideBar/TrackItem.vue

```diff
@@ -0,0 +1,373 @@
+            :markers="1"
+            selectionColor="transparent"
+            :disable="uiLocked"
+            @change="setTrackPan($event)"
+            @dblclick="setTrackPan(0)"
+          />
+          <div class="r">R</div>
+        </div>
+        <div class="gain">
+          <QIcon name="volume_down" class="l" size="1rem" />
+          <QSlider
+            :modelValue="track.gain"
+            :min="0"
+            :max="1"
+            :step="0.01"
+            :markers="1"
+            :disable="uiLocked"
+            @change="setTrackGain($event)"
+            @dblclick="setTrackGain(1)"
+          />
+          <QIcon name="volume_up" class="r" size="1rem" />
+        </div>
+      </div>
+    </QItem>
+  </div>
+</template>
+<script setup lang="ts">
+import { computed, watch, ref } from "vue";
+import CharacterSelectMenu from "@/components/Sing/CharacterMenuButton/CharacterSelectMenu.vue";
+import SingerIcon from "@/components/Sing/SingerIcon.vue";
+import { useStore } from "@/store";
+import ContextMenu from "@/components/Menu/ContextMenu.vue";
+import { shouldPlayTracks } from "@/sing/domain";
+import { CharacterInfo, StyleInfo, TrackId } from "@/type/preload";
+import { getOrThrow } from "@/helpers/mapHelper";
+
+const props = defineProps<{
+  trackId: TrackId;
+
+  draggableClass: string;
+}>();
+
+const store = useStore();
+const uiLocked = computed(() => store.getters.UI_LOCKED);
+const track = computed(() => {
+  const track = store.state.tracks.get(props.trackId);
+  if (!track) throw new Error(`Track not found: ${props.trackId}`);
+  return track;
+});
+
+const tracks = computed(() => store.state.tracks);
+const isThereSoloTrack = computed(() =>
+  [...tracks.value.values()].some((track) => track.solo),
+);
+const shouldPlayTrack = computed(() =>
+  getOrThrow(shouldPlayTracks(store.state.tracks), props.trackId),
+);
+
+const setTrackPan = (pan: number) => {
+  if (store.state.songUndoableTrackOptions.panAndGain) {
+    store.dispatch("COMMAND_SET_TRACK_PAN", { trackId: props.trackId, pan });
+  } else {
+    store.dispatch("SET_TRACK_PAN", { trackId: props.trackId, pan });
+  }
+};
+
+const setTrackGain = (gain: number) => {
+  if (store.state.songUndoableTrackOptions.panAndGain) {
+    store.dispatch("COMMAND_SET_TRACK_GAIN", {
+      trackId: props.trackId,
+      gain,
+    });
+  } else {
+    store.dispatch("SET_TRACK_GAIN", { trackId: props.trackId, gain });
+  }
+};
+
+watch(
+  () => track.value.name,
+  () => {
+    temporaryTrackName.value = track.value.name;
+  },
+);
+
+const temporaryTrackName = ref(track.value.name);
+
+const updateTrackName = () => {
+  if (temporaryTrackName.value === track.value.name) return;
+  if (temporaryTrackName.value === "") {
+    temporaryTrackName.value = track.value.name;
+    return;
+  }
+  setTrackName(temporaryTrackName.value);
+};
+
+const setTrackName = (name: string) => {
+  store.dispatch("COMMAND_SET_TRACK_NAME", { trackId: props.trackId, name });
+};
```

Comment:
（細かいけど）
ここ一緒にしちゃったほうがちょっと見通し良いかも。
trackName周り煩雑なので、少しでも。。
```suggestion
  store.dispatch("COMMAND_SET_TRACK_NAME", { trackId: props.trackId, name });
};
```